tools/build/scripts/utils/addPackageMetadata.js

```diff
@@ -18,8 +18,11 @@ const rootPackageJsonPath = require.resolve(rootPackageJson);
 //Find the package.json of the project and retrieve the version and repository URL
 const packageJsonPath = path.join(process.cwd(), "package.json");
 const packageJson = JSON.parse(FS.readFileSync(packageJsonPath));
-const version = packageJson.version;
-const repositoryUrl = packageJson.repository.url;
+const version = packageJson.version ?? '';
```

Comment:
the pkg version should always be defined afaik
if its missing throw an error instead?

Reply:
Internally debated that. Makes sense.

Reply:
Only for the version? Like just warning for the url?

Reply:
Both imo

Reply:
personally not a fan of IIFE, @ben-polinsky do you have an opinion on this?

Reply:
Yes for the most part, agree. I'd just check for the presence and throw if none.

Reply:
do you really need an extra fn to just call `throw new error` ?

Reply:
Ah, I felt like this way was cleaner. Other options were using IIFE or if statements since we can't `throw new error` when using the "??"

Reply:
Can we fix this once and for all and get this merged?

Reply:
Done